### ISSUE TITLE
Return workspace command results as json

### ISSUE THREAD

**Issue by erick-thompson** [hashicorp/terraform#20658]
### Current Terraform Version
```
Terraform v0.11.11
```

### Use-cases
I am building a set of scripts that use Terraform to provision a variety of cloud resources (Azure, but that isn't relevant). The script needs to invoke workspace commands, as the script needs to ensure that a particular workspace exists and is selected. Currently the output of the workspace command is text sent to stdout.

### Attempted Solutions
I've made some attempts at parsing the output of the workspace commands, but they are not especially friendly to parse. For example, workspace list returns a workspace per line, with the current workspace preceded by a *. Not hard to parse, but it relies on a very specific output format.

### Proposal
Terraform workspace commands should have a switch to return the results as json. For example, 

workspace list -json

might return 
```

[
    {
        "workspace": "foo",
        "selected": true
    },
    {
        "workspace": "bar",
        "selected": false
    }
]
```
### References
I didn't find any mentioned of workspace commands output format.

**Comment by SeanHil** [comment]
ma ata dafok

**Comment by hallvors** [comment]
I'm trying to use `terraform workspace list` output in a script. Well, it outputs workspaces (including that for scripting purposes inconvenient asterix) followed by several newlines and 

> The active workspace is being overridden using the TF_WORKSPACE environment variable

... argh!
